std/variant.d

```diff
@@ -1021,64 +1021,15 @@ public:
      * involved. The conversion rules mimic D's built-in rules for
      * arithmetic conversions.
      */
-
-    // Adapted from http://www.prowiki.org/wiki4d/wiki.cgi?DanielKeep/Variant
-    // arithmetic
-    VariantN opAdd(T)(T rhs) { return opArithmetic!(T, "+")(rhs); }
+    VariantN opBinary(string op, T)(T rhs)
+    if ((op == "+" || op == "-" || op == "*" || op == "/" || op == "^^") && 
+        is(typeof(opArithmetic!(T, op)(rhs))))
+    { return opArithmetic!(T, op)(rhs); }
     ///ditto
-    VariantN opSub(T)(T rhs) { return opArithmetic!(T, "-")(rhs); }
-
-    // Commenteed all _r versions for now because of ambiguities
-    // arising when two Variants are used
-
-    // ///ditto
-    // VariantN opSub_r(T)(T lhs)
-    // {
-    //     return VariantN(lhs).opArithmetic!(VariantN, "-")(this);
-    // }
-    ///ditto
-    VariantN opMul(T)(T rhs) { return opArithmetic!(T, "*")(rhs); }
-    ///ditto
-    VariantN opDiv(T)(T rhs) { return opArithmetic!(T, "/")(rhs); }
-    // ///ditto
-    // VariantN opDiv_r(T)(T lhs)
-    // {
-    //     return VariantN(lhs).opArithmetic!(VariantN, "/")(this);
-    // }
-    ///ditto
-    VariantN opMod(T)(T rhs) { return opArithmetic!(T, "%")(rhs); }
-    // ///ditto
-    // VariantN opMod_r(T)(T lhs)
-    // {
-    //     return VariantN(lhs).opArithmetic!(VariantN, "%")(this);
-    // }
-    ///ditto
-    VariantN opAnd(T)(T rhs) { return opLogic!(T, "&")(rhs); }
-    ///ditto
-    VariantN opOr(T)(T rhs) { return opLogic!(T, "|")(rhs); }
-    ///ditto
-    VariantN opXor(T)(T rhs) { return opLogic!(T, "^")(rhs); }
-    ///ditto
-    VariantN opShl(T)(T rhs) { return opLogic!(T, "<<")(rhs); }
-    // ///ditto
-    // VariantN opShl_r(T)(T lhs)
-    // {
-    //     return VariantN(lhs).opLogic!(VariantN, "<<")(this);
-    // }
-    ///ditto
-    VariantN opShr(T)(T rhs) { return opLogic!(T, ">>")(rhs); }
-    // ///ditto
-    // VariantN opShr_r(T)(T lhs)
-    // {
-    //     return VariantN(lhs).opLogic!(VariantN, ">>")(this);
-    // }
-    ///ditto
-    VariantN opUShr(T)(T rhs) { return opLogic!(T, ">>>")(rhs); }
-    // ///ditto
-    // VariantN opUShr_r(T)(T lhs)
-    // {
-    //     return VariantN(lhs).opLogic!(VariantN, ">>>")(this);
-    // }
+    VariantN opBinary(string op, T)(T rhs)
+    if ((op == "&" || op == "|" || op == "^" || op == ">>" || op == "<<" || op == ">>>") && 
```

Comment:
It looks like the style target fails because of trailing whitespace:
```
Running DScanner
generated/dscanner-e4ee46366d21eaa613da13c52f3f1a66684cdbaa/dsc --config .dscanner.ini --styleCheck etc std -I.
Check for trailing whitespace
grep -nr '[[:blank:]]$' etc std ; test $? -eq 1
std/variant.d:1025:    if ((op == "+" || op == "-" || op == "*" || op == "/" || op == "^^") &&
std/variant.d:1030:    if ((op == "&" || op == "|" || op == "^" || op == ">>" || op == "<<" || op == ">>>") &&
posix.mak:561: recipe for target 'style_lint' failed
make: *** [style_lint] Error 1
rm generated/linux/release/64/publictests/.directory
🚨 Error: The command exited with status 2
```